docs/howto/profiling.md

```diff
@@ -0,0 +1,153 @@
+# Profiling
+
+Software profiling allows you to inspect the performance of a piece of software, seeing where the bottlenecks are and how they can be improved.
+gperftools is a software profiling package. It contains a CPU profiler, thread-caching malloc library, memory leak detection tool, memory allocation profiler and pprof (discussed later). More information about gperftools and its components can be found on the project's [GitHub](https://github.com/gperftools/gperftools) and [documentation page](https://gperftools.github.io/gperftools/).
+
+## Install gperftools
+
+### Ubuntu
+
+If you're using Ubuntu you can use the following command to install gperftools:
+
+```
+apt install google-perftools libgoogle-perftools-dev
+```
+
+### Other Systems
+
+Alternatively, you can use the following commands to install it:
+
+```
+git clone https://github.com/gperftools/gperftools
+cd gperftools
+./autogen.sh
+./configure
+make
+make install

```

Comment:
Will this install in a location that ACTS CMake can find?
I guess that is for systems with root access. Is there a way to set it up in another directory?

Reply:
Added a build option for specifying the install directory of gperftools and instructions on how to specify a directory when installing gperftools.

Reply:
Tested this on Mercury and was able to profile ACTS.